devices/webcam/hand_tracking_utility.py

```diff
@@ -0,0 +1,600 @@
+import h5py
+import os
+import re
+import time
+import asyncio
+import labgraph as lg
+import numpy as np
+import mediapipe as mp
+import argparse as ap
+
+import mediapipe.python.solutions.hands as HandsType
+import mediapipe.python.solutions.drawing_utils as DrawingUtilsType
+import mediapipe.python.solutions.drawing_styles as DrawingStylesType
+
+from mediapipe.framework.formats.landmark_pb2 import NormalizedLandmarkList
+from typing import NamedTuple, Optional, Tuple, List
+from pathlib import Path
+
+from webcam import VideoFrame, WebcamStreamConfig, WebcamStream
+
+mp_drawing: DrawingUtilsType = mp.solutions.drawing_utils
+mp_drawing_styles: DrawingStylesType = mp.solutions.drawing_styles
+mp_hands: HandsType = mp.solutions.hands
+
+parser = ap.ArgumentParser()
+parser.add_argument("-te", "--test", help = "enable test mode", action = "store_true", required = False)
+parser.add_argument("-li", "--log-images", help = "add image data to logs", action = "store_true", required = False)
+parser.add_argument("-lp", "--log-poses", help = "add pose estimation data to logs", action = "store_true", required = False)
+parser.add_argument("-lf", "--log-format", type = str, nargs = 1, choices = ["hdf5"], help = "sepcify logging format. Supports hdf5, default is hdf5", action = "store", required = False)
+parser.add_argument("-ld", "--log-dir", type = str, nargs = 1, help = "sepcify log directory (must be full path), default is ./logs", action = "store", required = False)
+parser.add_argument("-ln", "--log-name", type = str, nargs = 1, help = "sepcify log filename, default is hand_tracking_utility", action = "store", required = False)
+parser.add_argument("-rp", "--replay", help = "replay a log file", action = "store_true", required = False)
+parser.add_argument("-dn", "--device-number", type = int, nargs = 1, help = "specify device number to capture from, default is 0", action = "store", required = False)
+parser.add_argument("-sr", "--sample-rate", type = int, nargs = 1, help = "specify sample rate from device in Hz, default is 30", action = "store", required = False)
+parser.add_argument("-dr", "--device-resolution", type = int, nargs = 2, help = "specify resolution for capture device, default is 854x480", action = "store", required = False)
+parser.add_argument("-ri", "--rescale-image", help = "enable rescaling display window from input size", action = "store_true", required = False)
+parser.add_argument("-rr", "--rescale_resolution", type = int, nargs = 2, help = "specify resolution to rescale display stream to, default is 1280x720", action = "store", required = False)
+
+class PoseFrame(lg.Message):
+    """
+    Represents a single frame of a webcam's stream along with its hand tracking estimations
+
+    @attributes:
+        frame: a (W, H, 3) shape array of the webcam's stream at its default resolution, in the blue/green/red color space
+        landmarks: List[NormalizedLandmarkList], one NormalizedLandmarkList object for each detected hand
```

Comment:
For users to understand more about NormalizedLandmarkList type, can you add an example and/or a reference to know more about the object?